engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -1359,7 +1409,8 @@ private void addDAGToCurrentDAG(ModuleMeta moduleMeta)
     String name;
     for (OperatorMeta operatorMeta : subDag.getAllOperators()) {
       name = subDAGName + MODULE_NAMESPACE_SEPARATOR + operatorMeta.getName();
-      this.addOperator(name, operatorMeta.getOperator());
+      Operator op = this.addOperator(name, operatorMeta.getOperator());
+      this.getMeta(op).copyFrom(operatorMeta);
       OperatorMeta operatorMetaNew = this.getOperatorMeta(name);
```

Comment:
Is operatormeta returned by `this.getMeta(op)` different from `this.getOperatorMeta(name)`? Can you not use `operatorMetaNew.copyFrom(operatorMeta)`?